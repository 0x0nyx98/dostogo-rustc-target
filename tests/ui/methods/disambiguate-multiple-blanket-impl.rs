trait A {
    fn foo(&self);
}

trait B {
    fn foo(&self);
}

#[derive(Debug)]
struct S;

impl<T: std::fmt::Debug> A for T {
    fn foo(&self) {} //~ NOTE candidate #1
}

impl<T: std::fmt::Debug> B for T {
    fn foo(&self) {} //~ NOTE candidate #2
}

fn main() {
    let s = S;
    S::foo(&s); //~ ERROR multiple applicable items in scope
    //~^ NOTE multiple `foo` found
    //~| HELP disambiguate
    //~| HELP disambiguate
}

