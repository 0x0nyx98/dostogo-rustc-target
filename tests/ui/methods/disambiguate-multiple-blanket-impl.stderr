error[E0034]: multiple applicable items in scope
  --> $DIR/disambiguate-multiple-blanket-impl.rs:22:8
   |
LL |     S::foo(&s);
   |        ^^^ multiple `foo` found
   |
note: candidate #1 is defined in an impl of the trait `A` for the type `T`
  --> $DIR/disambiguate-multiple-blanket-impl.rs:13:5
   |
LL |     fn foo(&self) {}
   |     ^^^^^^^^^^^^^
note: candidate #2 is defined in an impl of the trait `B` for the type `T`
  --> $DIR/disambiguate-multiple-blanket-impl.rs:17:5
   |
LL |     fn foo(&self) {}
   |     ^^^^^^^^^^^^^
help: disambiguate the method for candidate #1
   |
LL |     <T as A>::foo(&s);
   |     ~~~~~~~~~~
help: disambiguate the method for candidate #2
   |
LL |     <T as B>::foo(&s);
   |     ~~~~~~~~~~

error: aborting due to previous error

For more information about this error, try `rustc --explain E0034`.
